adding-autodeployment/README.md

```diff
@@ -0,0 +1,100 @@
+- cf install-plugin autopilot -f -r CF-Community
+```
+
+### deploy
+In order to make use of conditional deployments to staging/master on different branches, add the following to the `deploy` section of the `.travis.yml`:
+```
+- provider: script
+  script: "./deploy.sh staging"
+  skip_cleanup: true
+  on:
+    branch: develop
+- provider: script
+  script: "./deploy.sh production"
+  skip_cleanup: true
+  on:
+    branch: master
+```
+For more on conditional deployments, you can check out the [cloud.gov documentation](https://cloud.gov/docs/apps/continuous-deployment/#using-conditional-deployments).
+
+## 3. Add `deploy.sh`
+To account for the branch-wise logic that is generated in the `.travis.yml`, we will be using a separate `deploy.sh` shell script that will be called by travis at the time of deployment.
+
+Follow the comments to adjust the script to suit your needs as you see fit. You will need to update the `spaces` to account for what is available on your particular cloud.gov.
+
+```
+set -e
+
+API="https://api.fr.cloud.gov" # This will be the following url if you are still deploying to E/W 'https://api.cloud.gov'
+ORG="gsa-acq-proto"            #Or whatever cloud.gov org you wish to place the app in
+APP_NAME=""                    # Enter your app name
+
+SPACE=$1
+
+if [ $# -ne 1 ]; then
+  echo "Usage: deploy <space>"
+  exit
+fi
+
+if [ $SPACE = 'production' ]; then
+  NAME=APP_NAME
+  MANIFEST="manifest.yml"
+elif [ $SPACE = 'staging' ]; then
+  NAME="$APP_NAME-staging"
+  MANIFEST="manifest-staging.yml"
+else
+  echo "Unknown space: $SPACE"
+  exit
+fi
+
+cf login -a $API -u $CF_USERNAME -p $CF_PASSWORD -o $ORG -s $SPACE
+cf zero-downtime-push $NAME -f $MANIFEST
+```
+
+## 4. Add manifests
+Cloud.gov (and Cloud Foundry) use manifest files to specify how an app should be built on cloud.gov. You will now add two separate files, a `manifest.yml` for your production app and a `manifest-staging.yml` for your development application.
+
+Generally your production application will have multiple instances while your staging will only have one. Manifests can be short and sweet, or extensive. For the full cloud foundry documentation on manifests see here: https://docs.cloudfoundry.org/devguide/deploy-apps/manifest.html#minimal-manifest.
+
+For an example manifest and manifest-staging see here:
+[Acquisitions Manifest](https://github.com/18F/acquisitions.18f.gov/blob/develop/manifest.yml)
+[Acquisitions Manifest-Staging](https://github.com/18F/acquisitions.18f.gov/blob/develop/manifest-staging.yml)
+
+### Additional Considerations
+Your deployer credentials will regularly expire, so please make sure to update them periodically.
+
+Also, if your application successfully deploys to cloud.gov but does not start, which may happen for an application that does not have an adequate test suite, you may have to go into the cf target space and manually delete the "APP_NAME-venerable" application in order to make use of `autopilot` again.
```

Comment:
I've run into this issue a number of times, too. I wonder whether it's safe to have deletion of the `-venerable` app be the first step of the above deploy script.